R/geefSoortenlijst.R

```diff
@@ -55,20 +77,49 @@ geefSoortenlijst <-
     if (SoortengroepIDs$SoortengroepIDs == "") {
       stop("Voor de opgegeven argumenten is er geen soortenlijst")
     }
-
+  
     Soortenlijst <-
       geefSoortenlijstVoorIDs(
         Taxongroeplijst = SoortengroepIDs$SoortengroepIDs,
         Taxonlijsttype = Taxonlijsttype,
         ConnectieLSVIhabitats
       )
-
+  
     #soortgegevens aan selectiegegevens plakken
     SoortenlijstSelectie <- Selectiegegevens %>%
       left_join(
         Soortenlijst,
         by = ("TaxongroepId")
       )
+    
+    if (Taxonlijstniveau[1] == "voorwaarde") {
+      SoortenlijstSelectie <- SoortenlijstSelectie %>%
+        select(
+          -.data$Id
+        )
+    } else {
+      SoortenlijstSelectie <- SoortenlijstSelectie %>%
+        select(
+          .data$Versie, .data$Habitattype, .data$Habitatsubtype,
+          .data$Criterium, .data$Indicator, .data$Omschrijving,
+          .data$NbnTaxonVersionKey, .data$WetNaam, .data$NedNaam,
+          .data$WetNaamKort, .data$TaxonType
+        ) %>%
+        distinct()
+    }
+
+    if (Taxonlijstniveau[1] == "criterium") {
+      SoortenlijstSelectie <- SoortenlijstSelectie %>%
+        select(-.data$Indicator) %>%
+        filter(!is.na(.data$NbnTaxonVersionKey)) %>%
```

Comment:
Hier voeg je `filter(!is.na(.data$NbnTaxonVersionKey)) %>%` toe.  Is dit ook nodig op regels 99 en 108?